library/core/src/slice/sort/select.rs

```diff
@@ -298,3 +314,26 @@ fn median_idx<T, F: FnMut(&T, &T) -> bool>(
     }
     b
 }
+
+// It's possible to re-use the insertion sort in the smallsort module, but with optimize_for_size it
+// would clutter that module with cfg statements and make it generally harder to read and develop.
+// So to decouple things and simplify it, we use a an even smaller bubble sort.
+#[cfg(feature = "optimize_for_size")]
+fn bubble_sort<T, F: FnMut(&T, &T) -> bool>(v: &mut [T], is_less: &mut F) {
+    let mut n = v.len();
+    let mut did_swap = true;
+
+    while did_swap && n > 1 {
+        did_swap = false;
+        for i in 1..n {
+            // SAFETY: The loop construction implies that `i` and `i - 1` will always be in-bounds.
+            unsafe {
+                if is_less(v.get_unchecked(i), v.get_unchecked(i - 1)) {
+                    v.swap_unchecked(i - 1, i);
+                    did_swap = true;
```

Comment:
Did you consider the "last swap" optimization instead? (mentioned [here](https://en.wikipedia.org/wiki/Bubble_sort#Optimizing_bubble_sort))

Reply:
Good idea, I think that shouldn't change binary-size much, let me try it.

Reply:
Together with `ptr::swap_nonoverlapping` we can cut it down from [38](https://godbolt.org/z/rzGcsMvds) instructions to [28](https://godbolt.org/z/4n7Eqe4YP). `ptr::swap_nonoverlapping` is also better for larger types, plus not having to call a function on each loop iteration is better.